### PR TITLE
Add a check for TPU Spawn barrrier

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Add a check for TPU Spawn barrrier

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/plugins/training_type/tpu_spawn.py [hunk 1]
```diff
@@ -170,7 +170,8 @@ def model_to_device(self) -> None:
         self.model = self.wrapped_model.to(self.device)
 
     def barrier(self, name: Optional[str] = None) -> None:
-        rendezvous(name)
+        if tpu_distributed:
+            rendezvous(name)
```
Comment (awaelchli) [inline pytorch_lightning/plugins/training_type/tpu_spawn.py:174]: so that's the equivalent of `torch.distributed.is_avalable() and torch.distributed.is_initialized()` correct?

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]
lgtm, just wait for the Test to pass

File: pytorch_lightning/plugins/training_type/tpu_spawn.py (hunk 1 above)
Comment (kaushikb11) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:174]: It's checking the world size.

```python
    if _TPU_AVAILABLE:
        return xm.xrt_world_size() > 1
    return False
```

Comment (ananthsub) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:174]: should we move these out of the training type plugins and into a general utilities? we can check if TPU or if torch distributed and handle accordingly

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/tpu_spawn.py (hunk 1 above)
Comment (kaushikb11) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:174]: I like the idea of having it in Training Type plugins.
